common/utils/index.js

```diff
@@ -745,17 +747,27 @@ function pushServicePlanToApiServer() {
   }
 }
 
-function registerInterOperatorCrds() {
+function registerSFEventsCrd() {
   const eventmesh = require('../../data-access-layer/eventmesh');
   return Promise.all([
-    eventmesh.apiServerClient.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR, CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICES),
-    eventmesh.apiServerClient.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR, CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_PLANS),
-    eventmesh.apiServerClient.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR, CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICEINSTANCES),
-    eventmesh.apiServerClient.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR, CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICEBINDINGS),
     eventmesh.apiServerClient.registerCrds(CONST.APISERVER.RESOURCE_GROUPS.INSTANCE, CONST.APISERVER.RESOURCE_TYPES.SFEVENT)
   ]);
 }
 
+function waitWhileCRDsAreRegistered() {
+  // We assume here that the Helm chart based deployment or the pre-start script of interoperator job in case of BOSH based deployment will take care of registering the CRDs. While it is not registered, we wait!
+  const eventmesh = require('../../data-access-layer/eventmesh');
+  logger.info('Checking if the CRDs are already registered');
+  return Promise.all([
+    eventmesh.apiServerClient.getCustomResourceDefinition(CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICES + '.' + CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR),
+    eventmesh.apiServerClient.getCustomResourceDefinition(CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_PLANS + '.' + CONST.APISERVER.RESOURCE_GROUPS.INTEROPERATOR)
```

Comment:
Why is SFServiceInstance and SFServiceBinding CRDs not waited for?

Reply:
Actually the wait for the `SFplans` and `SFServices` are needed because the next call tries to fetch them and produce catalog. However, `SFServiceInstance` and `SFServiceBindings` are not immediately processed. Only during service instance creation/binding, they are processed. Hence we are not waiting for them.

@amitmalav WDYT?